src/dec/xed-agen.c

```diff
@@ -38,7 +38,7 @@ xed_error_enum_t xed_agen(xed_decoded_inst_t* xedd,
     xed_uint64_t index_value = 0;
     xed_uint64_t segment_base = 0;
     xed_uint64_t scale  = 0;
-    xed_int64_t displacement = 0;
+    xed_uint64_t displacement = 0;
```

Comment:
Memory displacement is a signed value. Why did you change it here?

Intel SDM:
![image](https://github.com/intelxed/xed/assets/87116213/95c5a26b-13bc-4216-97a5-18f76255eeba)

Reply:
The signedness of displacement here may not matter as the value it holds will be promoted to an unsigned type when adding to an unsigned value like base.  However, it could be a potential bug if a narrower signed type is assigned to it.  Luckily it is assigned only once with and xed_decoded_inst_get_memory_displacement() returns init64_t so no signed extension is involved.  I can change this commit so that all displacement variable are signed type.

Reply:
I have updated the commit to not change all displacement variables to unsigned types.